src/Firebase/GetHackathonEvents.tsx

```diff
@@ -2,7 +2,7 @@ import { getDocs, collection } from "firebase/firestore";
 import { db } from "./FirebaseConfig";
 
 export const fetchHackathonEvents = async () => {
-  const colRef = collection(db, "hackathonEvents");
+  const colRef = collection(db, "In_hackathonEvents");
   const querySnapshot = await getDocs(colRef);
```

Comment:
Since we are using the document reference id as a foreign key, instead of storing it as another entry inside (id: docId) each document in 'hackathonEvents' we can get it directly.

This is a way:
`export const fetchHackathonEvents = async () => {
  const colRef = collection(db, "hackathonEvents");
  const querySnapshot = await getDocs(colRef);
  
  const events = querySnapshot.docs.reduce((acc, doc) => {
      acc[doc.id] = doc.data();
      return acc;
  }, {});

  return events;
};`

And in the hackathonEventsPage, you will probably have to map through the returned data after calling fetchHackathonEvents()